cpu/esp32/periph/hwrng.c

```diff
@@ -23,33 +23,9 @@
 #include "periph_conf.h"
 #include "periph/hwrng.h"
 
-static const uint32_t* RNG_DATA_REG = (uint32_t*)0x3ff75144;
+#define RNG_DATA_REG (*(volatile uint32_t *)0x3ff75144)
```

Comment:
This fixes by the way a bug. In the hope that others point me to bugs introduced by me so that I can learn from that, I try to be a good example:

@gschorcht: Without the `volatile` the compiler assumes the memory location is regular memory. The compiler is therefore allowed to assume that two successive reads from that location will return the same value, like it would on normal memory. Likely, the compiler would generate code that reads the value into a register and fills the buffer in `buf` with that exact same value over and over again. So at most 32 bit of randomness are copied to `buf`.

Reply:
@maribu Thanks for pointing that. In case of ESP32 it seems not to be a problem. `hwrng_read` always returns different values so that successive reads seem to return different values.